base/compiler/abstractinterpretation.jl

```diff
@@ -1956,30 +2003,30 @@ function collect_argtypes(interp::AbstractInterpreter, ea::Vector{Any}, vtypes::
     return argtypes
 end
 
-function abstract_eval_statement(interp::AbstractInterpreter, @nospecialize(e), vtypes::VarTable, sv::InferenceState)
-    if !isa(e, Expr)
-        if isa(e, PhiNode)
-            rt = Union{}
-            for val in e.values
-                rt = tmerge(rt, abstract_eval_special_value(interp, val, vtypes, sv))
-            end
-            return rt
-        end
-        return abstract_eval_special_value(interp, e, vtypes, sv)
-    end
-    e = e::Expr
+struct RTEffects
+    rt
+    effects::Effects
+    RTEffects(@nospecialize(rt), effects::Effects) = new(rt, effects)
+end
```

Comment:
This is unnecessary (`effects` returned from `abstract_eval_statement_expr` is never used), but rather increases the complexity of our code base. Let's remove it.

Reply:
We do use the `effects` from `abstract_eval_statement_expr` here: https://github.com/JuliaLang/julia/pull/44803/files/49e224cab01a2a65600e166ab8c8b8b5e0925125#diff-cf06383638884136da67baacb3cec2ed1b9cf52ac3443f783a999a58184f890aR185-R186

Well, currently its a dead branch, but presumably we do want to be able to access these effects inside of semi concrete eval as part of a follow up PR